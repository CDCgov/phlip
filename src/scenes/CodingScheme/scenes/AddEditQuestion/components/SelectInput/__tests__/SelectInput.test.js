import React from 'react'
import { SelectInput } from '../index'
import { shallow } from 'enzyme'

const props = {
  canModify: true,
  name: 'possibleAnswers',
  label: 'select-label',
  answerType: 3,
  type: 'text',
  input: {},
  classes: {},
  index: 1,
  currentValue: {},
  meta: {
    active: true,
    touched: true,
    error: false,
    warning: false
  },
  handleDelete: jest.fn(),
  handleUp: jest.fn(),
  handleDown: jest.fn(),
  fields: [],
  isEdit: false
}

describe('AddEditQuestion form -- SelectInput', () => {
  test('should render correctly', () => {
    expect(shallow(<SelectInput {...props} />)).toMatchSnapshot()
  })
})
