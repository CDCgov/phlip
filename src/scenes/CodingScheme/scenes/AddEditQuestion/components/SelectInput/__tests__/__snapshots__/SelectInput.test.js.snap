// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEditQuestion form -- SelectInput should render correctly 1`] = `
<FlexGrid
  align="stretch"
  circular={false}
  container={true}
  flex={false}
  justify="stretch"
  padding={0}
  raised={false}
  style={Object {}}
  type="row"
>
  <FlexGrid
    align="stretch"
    circular={false}
    container={false}
    flex={false}
    justify="stretch"
    padding={0}
    raised={false}
    style={
      Object {
        "marginTop": 8,
      }
    }
    type="column"
  >
    <WithStyles(Checkbox)
      disabled={true}
    />
  </FlexGrid>
  <FlexGrid
    align="stretch"
    circular={false}
    container={false}
    flex={true}
    justify="stretch"
    padding={0}
    raised={false}
    style={Object {}}
    type="column"
  >
    <WithStyles(FormControl)
      error={false}
      fullWidth={true}
    >
      <WithStyles(InputLabel)
        htmlFor="possibleAnswers"
        required={false}
        shrink={true}
      >
        select-label
      </WithStyles(InputLabel)>
      <WithStyles(Input)
        classes={
          Object {
            "disabled": undefined,
          }
        }
        disabled={false}
        id="possibleAnswers"
        type="text"
      />
    </WithStyles(FormControl)>
  </FlexGrid>
  <FlexGrid
    align="center"
    circular={false}
    container={true}
    flex={false}
    justify="stretch"
    padding={0}
    raised={false}
    style={Object {}}
    type="row"
  >
    <FlexGrid
      align="stretch"
      circular={false}
      container={true}
      flex={false}
      justify="stretch"
      padding={0}
      raised={false}
      style={Object {}}
      type="column"
    >
      <IconButton
        aria-label="Move answer choice up one position"
        color="action"
        disableRipple={false}
        disabled={false}
        iconSize={36}
        id="move-answer-1-up"
        onClick={[MockFunction]}
        tooltipText=""
      >
        arrow_drop_up
      </IconButton>
      <IconButton
        aria-label="Move answer choice down one position"
        color="action"
        disableRipple={false}
        disabled={false}
        iconSize={36}
        id="move-answer-1-down"
        onClick={[MockFunction]}
        style={
          Object {
            "marginTop": -16,
          }
        }
        tooltipText=""
      >
        arrow_drop_down
      </IconButton>
    </FlexGrid>
    <FlexGrid
      align="stretch"
      circular={false}
      container={false}
      flex={false}
      justify="stretch"
      padding={0}
      raised={false}
      style={Object {}}
      type="column"
    >
      <IconButton
        aria-label="Delete 1 answer"
        color="action"
        disabled={false}
        iconSize={23}
        id="delete-answer-1"
        onClick={[MockFunction]}
        style={
          Object {
            "height": "20px !important",
          }
        }
        tooltipText=""
      >
        delete
      </IconButton>
    </FlexGrid>
  </FlexGrid>
</FlexGrid>
`;
